src/main/java/uk/gov/hmcts/sptribs/notification/NotificationHelper.java

```diff
@@ -90,16 +91,25 @@ public NotificationRequest buildLetterNotificationRequest(Map<String, Object> te
 
     public void setRecordingTemplateVars(Map<String, Object> templateVars, RecordListing recordListing) {
         templateVars.put(CommonConstants.CIC_CASE_HEARING_TYPE, recordListing.getHearingType());
-        templateVars.put(CommonConstants.CIC_CASE_HEARING_DATE, recordListing.getHearingDate());
+
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(CommonConstants.CIC_CASE_UK_DATE_FORMAT);
+        templateVars.put(CommonConstants.CIC_CASE_HEARING_DATE, recordListing.getHearingDate().format(dateTimeFormatter));
         templateVars.put(CommonConstants.CIC_CASE_HEARING_TIME, recordListing.getHearingTime());
 
-        if (null != recordListing.getSelectedVenue()) {
-            templateVars.put(CommonConstants.CIC_CASE_HEARING_VENUE, recordListing.getSelectedVenue());
-        } else {
+        if (isVideoFormat(recordListing) || isTelephoneFormat(recordListing)) {
             templateVars.put(CommonConstants.CIC_CASE_HEARING_VENUE, CommonConstants.CIC_CASE_RECORD_REMOTE_HEARING);
-        }
-        if (null != recordListing.getImportantInfoDetails()) {
-            templateVars.put(CommonConstants.CIC_CASE_HEARING_INFO, recordListing.getImportantInfoDetails());
+        } else
+            if (null != recordListing.getSelectedVenue()) {
+                templateVars.put(CommonConstants.CIC_CASE_HEARING_VENUE, recordListing.getSelectedVenue());
```

Comment:
This if/else logic is getting convoluted and a little hard to read.
Would it be possible to simplify it a bit by using a switch statement instead?

Reply:
Had discussion with Fergal why it can't be in switch.